bin/citrea/tests/bitcoin_e2e/light_client_test.rs

```diff
@@ -143,3 +149,317 @@ async fn test_light_client_proving() -> Result<()> {
+
+        // Finalize the DA block which contains the commitment txs
+        da.generate(FINALITY_DEPTH, None).await.unwrap();
+
+        let commitment_l1_height = da.get_finalized_height().await.unwrap();
+
+        // Wait for batch prover to generate proofs for commitments
+        batch_prover
+            .wait_for_l1_height(commitment_l1_height, Some(TEN_MINS))
+            .await
+            .unwrap();
+
+        // Assert that commitments are queryable
+        let commitments = batch_prover
+            .client
+            .http_client()
+            .get_sequencer_commitments_on_slot_by_number(commitment_l1_height)
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(commitments.len(), 1);
+
+        // Ensure that batch proofs is submitted to DA (1x reveal & 1x commit txs)
+        da.wait_mempool_len(2, Some(TEN_MINS)).await.unwrap();
+
+        // Finalize the DA block which contains the batch proof tx
+        da.generate(FINALITY_DEPTH, None).await?;
+        let batch_proof_l1_height = da.get_finalized_height().await?;
+        // Wait for the full node to see all process verify and store all batch proofs
+        full_node
+            .wait_for_l1_height(batch_proof_l1_height, Some(TEN_MINS))
+            .await?;
+        let batch_proofs = wait_for_zkproofs(
+            full_node,
+            batch_proof_l1_height,
+            Some(Duration::from_secs(30)),
+        )
+        .await
+        .unwrap();
+        assert_eq!(batch_proofs.len(), 1);
+
+        // Wait for light client prover to process batch proofs.
+        light_client_prover
+            .wait_for_l1_height(batch_proof_l1_height, Some(TEN_MINS))
+            .await
+            .unwrap();
+
+        // Expect light client prover to have generated light client proof
+        let lcp3 = light_client_prover
+            .client
+            .http_client()
+            .get_light_client_proof_by_l1_height(batch_proof_l1_height)
+            .await?;
+        assert!(lcp3.is_some());
+
+        let light_client_proof3 = lcp3.unwrap();
+        assert_eq!(
+            light_client_proof3
+                .light_client_proof_output
+                .state_root
+                .to_vec(),
+            batch_proofs[0].proof_output.final_state_root
+        );
+
+        assert_eq!(
+            light_client_proof3
+                .light_client_proof_output
+                .l2_genesis_state_root,
+            light_client_proof
+                .light_client_proof_output
+                .l2_genesis_state_root
+        );
+
+        assert_ne!(
+            light_client_proof3.light_client_proof_output.last_l2_height,
+            light_client_proof.light_client_proof_output.last_l2_height
+        );
+
+        assert_ne!(
+            light_client_proof3.light_client_proof_output.state_root,
+            light_client_proof.light_client_proof_output.state_root
+        );
+
```

Comment:
can you check if any non-stitchable batch proof tuples were put into the output vec?